### PR TITLE
Stop pretending to use require (bug 1113801)

### PR DESCRIPTION
**Pull request by kumar303**
The old require() stub was very node-like which
means it is not compatible with RequireJS and this
caused problems when trying to plug fxpay into
Fireplace.

This patch isn't ideal but maybe we can fix it later
by doing proper module-fi-cation. All the patch does
is stop making namespace checks look like calls to
require() since they're not.

@muffinresearch r?

### CODE REVIEW

**Comment by kumar303** [comment]
I would agree to that if Javascript didn't suck balls. This causes no error:

    var products = fxpay.products;

Until sometime later in the code you try:

    products.all();

which is generally too late to be helpful.

Maybe I should change it to:

    var products = fxGetGlobal('fxpay.products');

?

I realize now that the slashes are dumb anyway so I'll fix those at least.

**Comment by kumar303** [comment]
@muffinresearch you gave me some ideas to simplify it further :) Take another look when you get a sec. We still may want to hook properly into some module loader system because it will buy us some things.

I haven't forgotten about fixing the exports thing. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1119842

File: lib/fxpay/init_module.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+(function() {
+  'use strict';
+
+  if (typeof window.fxpay !== 'undefined') {
+    throw new Error('fxpay/init_module.js must be loaded before other scripts');
+  }
+
+  var exports = window.fxpay = {};
+
+  exports.getattr = function getattr(attr) {
```
Comment (muffinresearch) [inline lib/fxpay/init_module.js]: As I mentioned on the call would be nice to maybe make use of `obj.hasOwnProperty(prop)` here so we're not looking up things down the prototype chain.

**Comment by muffinresearch** [comment]
lgtm r+wc

File: lib/fxpay/init_module.js (hunk 1 above)
Comment (kumar303) [inline lib/fxpay/init_module.js]: Done. Thanks for the r.

**Pull request merged**